crates/egui/src/load.rs

```diff
@@ -274,17 +314,27 @@ pub trait TextureLoader {
 }
 
 #[derive(Default)]
-pub(crate) struct IncludeBytesLoader {
-    cache: Mutex<HashMap<&'static str, Arc<[u8]>>>,
+pub(crate) struct DefaultBytesLoader {
+    cache: Mutex<HashMap<&'static str, Bytes>>,
 }
 
-impl IncludeBytesLoader {
-    pub(crate) fn insert(&self, uri: &'static str, bytes: &'static [u8]) {
-        self.cache.lock().entry(uri).or_insert_with(|| bytes.into());
+impl DefaultBytesLoader {
+    pub(crate) fn insert_static(&self, uri: &'static str, bytes: &'static [u8]) {
+        self.cache
+            .lock()
+            .entry(uri)
+            .or_insert_with(|| Bytes::Static(bytes));
+    }
+
+    pub(crate) fn insert_shared(&self, uri: &'static str, bytes: impl Into<Arc<[u8]>>) {
```

Comment:
```suggestion
    pub(crate) fn insert_static(&self, uri: &'static str, bytes: impl Into<Bytes>) {
        self.cache
            .lock()
            .entry(uri)
            .or_insert_with(|| bytes.into());
    }
```

Reply:
There's a tiny optimization here: `Bytes` requires `Arc<[u8]>` to already be constructed, `impl Into<Arc<[u8]>>` doesn't. So here the user would have to copy their bytes into a `Arc<[u8]>` first if the data is not `'static`, where as with this split, we only copy them if there isn't already an entry for the given `uri`.

Is it worth keeping?